tests/base/src/test/java/org/keycloak/tests/admin/event/AdminEventTest.java

```diff
@@ -0,0 +1,433 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.admin.client.resource.UserResource;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.admin.AdminEvent;
+import org.keycloak.events.admin.AuthDetails;
+import org.keycloak.events.admin.OperationType;
+import org.keycloak.events.admin.ResourceType;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.AdminEventRepresentation;
+import org.keycloak.representations.idm.AuthDetailsRepresentation;
+import org.keycloak.representations.idm.OrganizationDomainRepresentation;
+import org.keycloak.representations.idm.OrganizationRepresentation;
+import org.keycloak.representations.idm.RealmEventsConfigRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testframework.annotations.InjectAdminClient;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.events.AdminEventAssertion;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.RealmConfig;
+import org.keycloak.testframework.realm.RealmConfigBuilder;
+import org.keycloak.testframework.realm.UserConfigBuilder;
+import org.keycloak.testframework.remote.runonserver.InjectRunOnServer;
+import org.keycloak.testframework.remote.runonserver.RunOnServerClient;
+import org.keycloak.tests.utils.admin.ApiUtil;
+import org.keycloak.util.JsonSerialization;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Test getting and filtering admin events.
+ *
+ * @author Stan Silvert ssilvert@redhat.com (C) 2016 Red Hat Inc.
+ */
+@KeycloakIntegrationTest
+public class AdminEventTest {
+
+    @InjectRealm(attachTo = "master", ref = "master")
+    ManagedRealm masterRealm;
+
+    @InjectRealm(config = AdminEventRealmConfig.class, ref = "default")
+    ManagedRealm managedRealm;
+
+    @InjectAdminClient(mode = InjectAdminClient.Mode.BOOTSTRAP)
+    Keycloak adminClient;
+
+    @InjectRunOnServer(realmRef = "default")
+    RunOnServerClient runOnServerClient;
+
+
+    @BeforeEach
+    public void initConfig() {
+        managedRealm.admin().clearAdminEvents();
+    }
+
+    private List<AdminEventRepresentation> events() {
+        return managedRealm.admin().getAdminEvents();
+    }
+
+    private String createUser(String username) {
+        UserRepresentation user = UserConfigBuilder.create()
```

Comment:
@vaceksimon @stianst @miquelsi  Don't we want to introduce a util class instead, not mixing test util classes and framework helper classes? To avoid boiler plating we can create UserConfigBuilderUtil class (placed in the utils module) which will extend UserConfigBuilder class? WDYT?

Reply:
This is probably quite a common pattern that you need to create for example a user, and obtain the returned UUID. Could be something part of the core framework I'd say. Not quite sure how, but create an issue for it and we can discuss it as a follow-up.

Reply:
I agree, let's leave it as it is : +1